lib/standards/aria-roles.js

```diff
@@ -0,0 +1,221 @@
+// Source: https://www.w3.org/TR/wai-aria-1.1/#roles
+
+/* easiest way to see allowed roles is to filter out the global ones
+   from the list of inherited states and properties. The dpub spec
+   does not have the global list so you'll need to copy over from
+   the wai-aria one:
+
+  const globalAttrs = Array.from(
+    document.querySelectorAll('#global_states li')
+  ).map(li => li.textContent.replace(/\s*\(.*\)/, ''));
+
+  const globalRoleAttrs = Array.from(
+    document.querySelectorAll('.role-inherited li')
+  ).filter(li => globalAttrs.includes(li.textContent.replace(/\s*\(.*\)/, '')))
+
+  globalRoleAttrs.forEach(li => li.style.display = 'none');
+*/
+const ariaRoles = {
+	alert: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded'],
+		nameFrom: ['author']
+	},
+	alertdialog: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded', 'aria-modal'],
+		nameFrom: ['author']
+	},
+	application: {
+		type: 'landmark',
+		allowedAttrs: ['aria-activedescendant'],
+		nameFrom: ['author']
+	},
+	article: {
+		type: 'structure',
+		allowedAttrs: ['aria-posinset', 'aria-setsize', 'aria-expanded'],
+		nameFrom: ['author'],
+		implicit: ['article']
+	},
+	banner: {
+		type: 'landmark',
+		allowedAttrs: ['aria-expanded'],
+		nameFrom: ['author'],
+		implicit: ['header']
+	},
+	button: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded', 'aria-pressed'],
+		nameFrom: ['contents', 'author'],
+		implicit: [
+			'button',
+			'input[type="button"]',
+			'input[type="image"]',
+			'input[type="reset"]',
+			'input[type="submit"]',
+			'summary'
+		]
+	},
+	cell: {
+		type: 'structure',
+		owned: ['row'],
+		allowedAttrs: [
+			'aria-colindex',
+			'aria-colspan',
+			'aria-rowindex',
+			'aria-rowspan',
+			'aria-expanded'
+		],
+		nameFrom: ['contents', 'author'],
+		implicit: ['td', 'th']
+	},
+	checkbox: {
+		type: 'widget',
+		requiredAttrs: ['aria-checked'],
+		// Note: aria-required is not in the 1.1 spec but is
+		// consistently supported in ATs and was added in 1.2.
+		allowedAttrs: ['aria-readonly', 'aria-required'],
+		nameFrom: ['contents', 'author'],
+		implicit: ['input[type="checkbox"]']
+	},
+	columnheader: {
+		type: 'structure',
+		context: ['row'],
+		allowedAttrs: [
+			'aria-sort',
+			'aria-colindex',
+			'aria-colspan',
+			'aria-expanded',
+			'aria-readonly',
+			'aria-required',
+			'aria-rowindex',
+			'aria-rowspan',
+			'aria-selected'
+		],
+		implicit: ['th']
+	},
+	combobox: {
+		type: 'composite',
+		owned: ['listbox', 'tree', 'grid', 'dialog', 'textbox'],
```

Comment:
How about we do this. That's what ARIA calls them.
```suggestion
		requiredOwned: ['listbox', 'tree', 'grid', 'dialog', 'textbox'],
```